### ISSUE TITLE
IndentationCheck: false positive with nested records' methods (Java 14 + Java 15)

### ISSUE THREAD

**Issue by vovkss** [checkstyle/checkstyle#8806]
Check documentation: https://checkstyle.org/config_misc.html#Indentation

```
/var/tmp $ javac OuterClass.java --enable-preview --release=15
Note: OuterClass.java uses preview language features.
Note: Recompile with -Xlint:preview for details.


/var/tmp $ cat config.xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
     "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="Indentation">
      <property name="basicOffset" value="4"/>
      <property name="braceAdjustment" value="0"/>
      <property name="caseIndent" value="4"/>
      <property name="throwsIndent" value="4"/>
      <property name="arrayInitIndent" value="4"/>
      <property name="lineWrappingIndentation" value="4"/>
      <property name="forceStrictCondition" value="false"/>
    </module>
  </module>
</module>

/var/tmp $ cat OuterClass.java
public final class OuterClass {

    private record InnerRecord(
        String value
    ) {
        @Override
        public boolean equals(final Object obj) {
            return
                obj instanceof InnerRecord theObj
                && value.equals(theObj.value);
        }

        @Override
        public int hashCode() {
            return value.hashCode();
        }
    }

}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-8.36-all.jar -c config.xml OuterClass.java
Starting audit...
[ERROR] C:\Users\vovks\Desktop\xx\OuterClass.java:6:9: 'method def modifier' has incorrect indentation level 8, expected level should be 4. [Indentation]
[ERROR] C:\Users\vovks\Desktop\xx\OuterClass.java:8:13: 'method def' child has incorrect indentation level 12, expected level should be 8. [Indentation]
[ERROR] C:\Users\vovks\Desktop\xx\OuterClass.java:11:9: 'method def rcurly' has incorrect indentation level 8, expected level should be 4. [Indentation]
[ERROR] C:\Users\vovks\Desktop\xx\OuterClass.java:13:9: 'method def modifier' has incorrect indentation level 8, expected level should be 4. [Indentation]
[ERROR] C:\Users\vovks\Desktop\xx\OuterClass.java:15:13: 'method def' child has incorrect indentation level 12, expected level should be 8. [Indentation]
[ERROR] C:\Users\vovks\Desktop\xx\OuterClass.java:16:9: 'method def rcurly' has incorrect indentation level 8, expected level should be 4. [Indentation]
Audit done.
Checkstyle ends with 6 errors.
```

---------------

This issue reproduces on both JDK 14 and on JDK 15 early-access build. 

My expectation was that no errors are emitted by Checkstyle in this case. 
If I replace the nested record with a nested static class, there are no errors. 

It seems that `IndentationCheck` is not aware of records and treats their contents as if they were the contents of the enclosing class.

**Comment by strkkk** [comment]
Fix is merged
